### PR TITLE
chore(webapp): migrate to vite 5

### PR DESCRIPTION
**Pull request by Rotzbua**
## Guide

https://vitejs.dev/guide/migration.html

## Changes

* Migrate to vite v5
* Change build to ESM
* Update CSP hash for legacy support
* Fix sass warnings and eslint error

### CODE REVIEW

**Comment by peterthomassen** [comment]
Closes #805

**Comment by Rotzbua** [comment]
@peterthomassen Works for me. Can you review? Happy xmas 🎄

File: www/webapp/package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "eslint-plugin-import": "^2.27.5",
     "eslint-plugin-vue": "^9.15.1",
     "eslint-plugin-vuetify": "^1.1.0",
-    "sass": "~1.69.5",
+    "sass": "~1.32.13",
```
Comment (peterthomassen) [inline www/webapp/package.json:36]: Why are we downgrading?

File: test/e2e2/spec/test_www.py [hunk 2]
```diff
@@ -88,12 +88,12 @@ def test_unknown_hosts(api_anon, protocol, hostname):
 def test_security_headers(api_anon):
     api_anon.headers = {}
     # CSP hashes are for legacy browser support.
-    # Source: https://github.com/vitejs/vite/tree/v4/packages/plugin-legacy#content-security-policy
+    # Source: https://github.com/vitejs/vite/tree/v5.2.0/packages/plugin-legacy#content-security-policy
```
Comment (peterthomassen) [inline test/e2e2/spec/test_www.py:91]: 404

**Review by peterthomassen — CHANGES_REQUESTED** [review]
Thanks! Looks mostly good; just minor stuff.

According to https://vitejs.dev/guide/migration#allow-path-containing-to-fallback-to-index-html, this also fixes #805. Can you confirm that's the case?

**Comment by peterthomassen** [comment]
> Happy xmas 🎄

Same to you, very much belated! And happy new year! :partying_face:

**Review by peterthomassen — APPROVED** [review]
lgtm, thanks a lot!

**Pull request merged**